File: src/tasks/cli/command/common/simple/SimpleArguments.java
package tasks.cli.command.common.simple;

import omnia.data.structure.List;
import tasks.model.ObservableTask;

public abstract class SimpleArguments {
  private final List<ObservableTask> tasks;

  protected SimpleArguments(List<ObservableTask> tasks) {
    this.tasks = tasks;
  }

  /** The list of tasks parsed from the command line. */
  protected List<ObservableTask> tasks() {
    return tasks;
  }
}

File: src/tasks/cli/command/common/simple/SimpleArguments.kt
package tasks.cli.command.common.simple

import omnia.data.structure.List
import tasks.model.Task

abstract class SimpleArguments protected constructor(private val tasks: List<Task>) {

  /** The list of tasks parsed from the command line.  */
  open fun tasks(): List<Task> {
    return tasks
  }
}